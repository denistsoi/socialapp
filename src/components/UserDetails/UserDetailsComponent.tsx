import React, { ReactElement } from "react";
import { View, Text } from "react-native";

import Address from "../Address/UserAddressComponent";
import Albums from "../Albums/UserAlbumsComponent";

import { UserDetails, Album, Post, Todo } from "../../types/resource";
import { UserDetailsNavigationProp } from "./../../types/navigation";

type Props = {
  navigation: UserDetailsNavigationProp;
  user: UserDetails;
  albums: Array<Album> | null;
  posts: Array<Post> | null;
  todos: Array<Todo> | null;
};

const UserDetailsComponent = ({
  navigation,
  user,
  albums,
  posts,
  todos
}: Props): ReactElement => {
  return (
    <View>
      <Address
        address={user.address}
        onPress={(): void => {
          navigation.navigate("UserAddress", { address: user?.address });
        }}
      />

      <View>
        <Text>Albums</Text>
        <Albums
          albums={albums}
          onPress={(albumId: string): void => {
            navigation.navigate("UserAlbum", { album: albumId });
          }}
        />
      </View>

      <Text>{JSON.stringify(todos)}</Text>

      <Text>{JSON.stringify(posts)}</Text>
    </View>
  );
};

export default UserDetailsComponent;
