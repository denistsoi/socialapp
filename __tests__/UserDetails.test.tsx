import React from "react";
import { render } from "react-native-testing-library";
import UserDetailsComponent from "../src/components/UserDetails/UserDetailsComponent";

describe("UserDetails", () => {
  it("should render a single user address details in UserDetailsComponent", async () => {
    let props: any;

    props = {
      navigation: {
        navigate: jest.fn()
      },
      ...props,
      user: {
        id: "1",
        name: "Bob",
        username: "Bob",
        address: {
          suite: "1/F",
          street: "101 Avenue"
        }
      }
    };

    const component = render(
      <UserDetailsComponent navigation={props.navigation} user={props.user} />
    );

    expect(component.toJSON()).toMatchSnapshot();
    expect(component.getByText("1/F 101 Avenue")).toBeDefined();
  });
});
