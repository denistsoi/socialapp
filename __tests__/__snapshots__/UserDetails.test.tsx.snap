// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserDetails should render a single user address details in UserDetailsComponent 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#efefef",
      "flex": 1,
      "height": "100%",
      "justifyContent": "center",
      "padding": 20,
      "paddingBottom": 40,
      "width": "100%",
    }
  }
>
  <RCTScrollView
    collapsable={false}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    style={
      Object {
        "backgroundColor": "#fff",
        "height": "100%",
        "width": "100%",
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "backgroundColor": "#74b9ff",
            "padding": 10,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 16,
                "justifyContent": "center",
              }
            }
          >
            Bob
          </Text>
          <Text
            style={
              Object {
                "fontSize": 12,
              }
            }
          >
            @
            Bob
          </Text>
        </View>
        <View>
          <RNGestureHandlerButton
            collapsable={false}
            onGestureEvent={[Function]}
            onGestureHandlerEvent={[Function]}
            onGestureHandlerStateChange={[Function]}
            onHandlerStateChange={[Function]}
            rippleColor={0}
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                undefined,
              ]
            }
          >
            <View
              accessible={true}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View>
                <Text
                  style={
                    Object {
                      "fontSize": 10,
                    }
                  }
                >
                  1/F
                   
                  101 Avenue
                </Text>
                <Text
                  style={
                    Object {
                      "fontSize": 10,
                    }
                  }
                >
                   
                </Text>
              </View>
            </View>
          </RNGestureHandlerButton>
        </View>
      </View>
      <View
        style={
          Object {
            "minHeight": 100,
            "paddingHorizontal": 10,
            "paddingVertical": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 16,
              "fontWeight": "bold",
              "paddingBottom": 4,
            }
          }
        >
          Albums
        </Text>
        <ActivityIndicator
          animating={true}
          color="#999999"
          hidesWhenStopped={true}
          size="small"
        />
      </View>
      <View
        style={
          Object {
            "minHeight": 100,
            "paddingHorizontal": 10,
            "paddingVertical": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 16,
              "fontWeight": "bold",
              "paddingBottom": 4,
            }
          }
        >
          Posts
        </Text>
        <ActivityIndicator
          animating={true}
          color="#999999"
          hidesWhenStopped={true}
          size="small"
        />
      </View>
      <View
        style={
          Object {
            "minHeight": 100,
            "paddingHorizontal": 10,
            "paddingVertical": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 16,
              "fontWeight": "bold",
              "paddingBottom": 4,
            }
          }
        >
          Todos
        </Text>
        <ActivityIndicator
          animating={true}
          color="#999999"
          hidesWhenStopped={true}
          size="small"
        />
      </View>
    </View>
  </RCTScrollView>
</View>
`;
